packages/plugin-rsc/src/plugin.ts

```diff
@@ -1273,15 +1285,94 @@ function generateDynamicImportCode(map: Record<string, string>) {
   return `export default {${code}};\n`
 }
 
-// // https://github.com/vitejs/vite/blob/2a7473cfed96237711cda9f736465c84d442ddef/packages/vite/src/node/plugins/importAnalysisBuild.ts#L222-L230
-function assetsURL(url: string) {
+class RuntimeAsset {
+  runtime: string
+  constructor(value: string) {
+    this.runtime = value
+  }
+}
+
+function serializeValueWithRuntime(value: any) {
+  const replacements: [string, string][] = []
+  let result = JSON.stringify(value, (_key, value) => {
+    if (value instanceof RuntimeAsset) {
+      const placeholder = `__runtime_placeholder_${replacements.length}__`
+      replacements.push([placeholder, value.runtime])
+      return placeholder
+    }
+
+    return value
+  })
+
+  for (let [placeholder, runtime] of replacements) {
+    result = result.replace(`"${placeholder}"`, runtime)
+  }
+
+  return result
+}
+
+function assetsURL(
+  url: string,
+  {
+    environment,
+    enableRuntimeValue = false,
+  }: {
+    environment: Environment
+    enableRuntimeValue?: boolean
+  },
+) {
+  if (
+    enableRuntimeValue &&
+    environment.mode === 'build' &&
+    typeof config.experimental?.renderBuiltUrl === 'function'
+  ) {
+    const result = config.experimental.renderBuiltUrl(url, {
+      type: 'asset',
+      hostType: 'js',
+      ssr: environment.name === 'ssr',
```

Comment:
Do you know what's the use case of switching something based on this `ssr` flag?

Currently, during the build, this is always `this.environment.name === 'client'` since assets manifest is generated during `generateBundle` of client build, so `ssr: false` is always false. Is it intended?

Reply:
I'm thinking the same. Let's hard-code `ssr: true` here, so we don't even need to pass `environment`.

Reply:
cool, making the change